File: 0134-gas-station/0134-gas-station.java
class Solution {
    public int canCompleteCircuit(int[] gas, int[] cost) {
        int totalDiff = 0, start = 0, currGas = 0;
        for(int i=0; i<gas.length; i++) {
            totalDiff += gas[i]-cost[i];
            currGas += gas[i]-cost[i];
            if(currGas < 0) {
                currGas = 0;
                start = i+1;
            }
        }
        return totalDiff >= 0? start: -1;
    }
}

File: 0134-gas-station/0134-gas-station.py
class Solution:
    def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
        if sum(gas) < sum(cost):
            return -1
        total_diff, start = 0, 0

        for i in range(len(gas)):
            total_diff += gas[i]-cost[i]

            if total_diff < 0:
                total_diff = 0
                start = i+1
        return start